chunky/src/java/se/llbit/chunky/renderer/scene/Scene.java

```diff
@@ -930,14 +933,25 @@ public synchronized void loadChunks(TaskTracker taskTracker, World world, Collec
         int wx0 = cp.x * 16; // Start of this chunk in world coordinates.
         int wz0 = cp.z * 16;
         BiomeData biomeData = chunkData.getBiomeData();
+        ChunkBiomeBlendingHelper chunkBiomeHelper = new ChunkBiomeBlendingHelper();
 
         if (use3dBiomes) {
-          for (int y = chunkData.minY(); y < chunkData.maxY(); y++) {
+          biomeBlendingHelper.put(cp, chunkBiomeHelper);
+          // We need to load biome data for the full height of the chunk
+          // and not only limited to the sections where there are blocks
```

Comment:
Can this be limited to `chunkData.minY() - blendRadius` … `chunkData.maxY() + blendRadius`?

Reply:
No, even if a chunk is full of air (so there are no block to load) its biome can affect neighboring chunks. Like you could have a leave bloc in a chunk at y=120 (near the border of a chunk) and have no clock higher than y=70 in the neighboring chunks, you still need to load the biome of the neighboring chunk at height around 120 to compute the correct biome color.
It was vital to load before I changed Position3d2IntPackedArray.java to return -1 in case of not found because before this change, it was impossible to make the difference between not biome information not loaded and biome loaded that happens to have index of 0 so we ended up blurring with a chunk that wasn't there. Now that we can detect, and ignore, unloaded biome during blur this isn't as vital because I won't produce obviously wrong result anymore but the correct way is still to fully load

Reply:
Oh yeah, that makes sense. Thanks for explaining it!